src/ml_tooling/transformers/pandas_transformers.py

```diff
@@ -34,18 +34,38 @@ def transform(self, X):
 # noinspection PyUnusedLocal
 class FillNA(BaseEstimator, TransformerMixin):
     """
-    Fills NA values with given value
+    Fills NA values with given value or strategy. If no value or strategy are
+    supplied missings are imputed with zero. If both a value and a strategy
+    are supplied the strategy will be used.
     """
 
-    def __init__(self, value):
+    def most_freq(X):
+        return pd.DataFrame.mode(X).iloc[0]
+
+    func_map_ = {'mean': pd.DataFrame.mean,
+                 'median': pd.DataFrame.median,
+                 'most_freq': most_freq,
```

Comment:
Why are we treating mode different?

Reply:
see [https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.mode.html](url):

> Note that there could be multiple values returned for the selected axis (when more than one item share the maximum frequency), which is the reason why a dataframe is returned. If you want to impute missing values with the mode in a dataframe df, you can just do this: df.fillna(df.mode().iloc[0])